src/trace/log.zig

```diff
@@ -17,19 +17,31 @@ pub const Logger = struct {
     exit_sig: AtomicBool,
     handle: ?std.Thread,
     channel: *Channel(*Entry),
+    entry_sink: EntrySink,
 
     const Self = @This();
 
-    pub fn init(allocator: std.mem.Allocator, default_level: Level) *Self {
+    pub fn init(allocator: std.mem.Allocator, default_level: Level, entry_sink: ?EntrySink) *Self {
         var self = allocator.create(Self) catch @panic("could not allocator.create Logger");
         var arena = std.heap.ArenaAllocator.init(allocator);
+
+        var sink = blk: {
```

Comment:
change to if comptime var on info/debug/... calls

Reply:
- seems you cant change comptime values on structs (eg, example below doesnt compile -- error: `value stored in comptime field does not match the default value of the field`) 
```zig 
const std = @import("std");
const builtin = @import("builtin");

const Foo = struct { 
  comptime do: bool = true,

  pub fn foo(self: @This()) void { 
    if (self.do) std.debug.print("hi", .{}); 
  }

  pub fn doNothing(self: *@This()) void { 
    self.do = false;
  }
};

pub fn main() void {
  var foo = Foo{}; 
  foo.foo(); 
  foo.doNothing(); 
  foo.foo(); 
}

```